numpy/core/numeric.py

```diff
@@ -1544,17 +1542,59 @@ def rollaxis(a, axis, start=0):
     return a.transpose(axes)
 
 
-def _validate_axis(axis, ndim, argname):
+def normalize_axis_tuple(axis, ndim, argname=None, allow_duplicate=False):
```

Comment:
I slightly wonder -- mostly because of #8819 -- whether it would be useful to have this in C...